apps/api/src/lesson/repositories/adminLesson.repository.ts

```diff
@@ -170,6 +172,66 @@ export class AdminLessonRepository {
       `);
   }
 
+  async getExistingQuestions(lessonId: UUIDType) {
+    return await this.db
+      .select({ id: questions.id })
+      .from(questions)
+      .where(eq(questions.lessonId, lessonId));
+  }
+
+  async upsertOptions(questionId: UUIDType) {
+    const existingOptions = await this.db
+      .select({ id: questionAnswerOptions.id })
+      .from(questionAnswerOptions)
+      .where(eq(questionAnswerOptions.questionId, questionId));
+
+    return { existingOptions };
+  }
+
+  async updateOption(optionId: UUIDType, optionData: OptionBody) {
+    await this.db
+      .update(questionAnswerOptions)
+      .set(optionData)
+      .where(eq(questionAnswerOptions.id, optionId));
+  }
+
+  async insertOption(questionId: UUIDType, optionData: OptionBody) {
+    await this.db.insert(questionAnswerOptions).values({
+      questionId,
+      ...optionData,
+    });
+  }
+
+  async deleteOptions(optionIds: UUIDType[]) {
+    await this.db
+      .delete(questionAnswerOptions)
+      .where(inArray(questionAnswerOptions.id, optionIds));
+  }
+
+  async deleteQuestions(questionsToDelete: UUIDType[]) {
+    await this.db.delete(questions).where(inArray(questions.id, questionsToDelete));
+  }
+
+  async deleteQuestionOptions(questionsToDelete: UUIDType[]) {
+    await this.db
+      .delete(questionAnswerOptions)
+      .where(inArray(questionAnswerOptions.questionId, questionsToDelete));
+  }
+
+  async upsertQuestion(questionData: QuestionBody, lessonId: UUIDType, authorId: UUIDType, questionId?: UUIDType): Promise<UUIDType> {
```

Comment:
Maybe we can use drizzle upsert to do it?
```suggestion
  await db.insert(users)
  .values({ id: 1, name: 'Dan' })
  .onConflictDoUpdate({ target: users.id, set: { name: 'John' } });
```